Run tests on GitHub actions

tvdeyen opened the pull request:
Using the https://github.com/solidusio/test-solidus-extension action to run tests on GitHub rather than CircleCI

Note: this also drops support for Rails 7.0 as it has several issues with Ruby >= 3.1 and is EOL since 2025-04-01

tvdeyen committed:
Run tests on GitHub actions

tvdeyen committed:
Add csv gem for Ruby 3.4 support

Ruby 3.4 extracted csv into a separate gem.
Solidus Promotions requires CSV but does not depend on
it yet.

tvdeyen committed:
Add GH Actions badge to README

fthobe commented:
Can we merge this before laying hand on this extension? I would like to bring this over and have it circleci free on the coming PR if we can.

tvdeyen committed:
Lock concurrent-ruby < 1.3.5 for Rails 7.0

https://github.com/rails/rails/issues/54271

tvdeyen committed:
Stop supporting Rails 7.0

Rails 7.0 has several issues with Ruby >= 3.1 and is EOL of live since 2025-04-01

tvdeyen committed:
Add rspec binstub

Is is common to run `bin/rspec` instead of `bundle exec rspec`.

tvdeyen committed:
TMP: Using solidus_dev_support branch that works with chrome 133

tvdeyen commented:
Needs https://github.com/solidusio/solidus_dev_support/pull/230 for latest chrome fixes

tvdeyen committed:
CI: Add Solidus 4.5 specs

And test Rails 8 with soliuds 4.5

tvdeyen committed:
CI: Runs feature specs with headless firefox

Chrome causes headaches

tvdeyen commented:
Needs https://github.com/solidusio/solidus/pull/6231
